xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift

```diff
@@ -800,12 +858,26 @@ struct UserProfileDetailModel: ModelProtocol {
     static func failFollow(
         state: Self,
         environment: Environment,
-        error: String
+        error: String,
+        context: FailFollowContext
     ) -> Update<Self> {
         var model = state
-        model.failFollowErrorMessage = error
+        model.isFollowSheetPresented = false
+        model.isFollowNewUserFormSheetPresented = false
         
-        return Update(state: model)
+        // We must defer between dismissing the sheet and presenting the alert
+        // or else SwiftUI throws a tantrum and swallows the alert
+        let fx: Fx<UserProfileDetailAction> = Future.detached {
```

Comment:
Q: if we dismiss the sheet immediately when beginning the action, is this still necessary?

It is a footgun lying is wait regardless, but we can document the other dismissal sites.